### ISSUE TITLE
[BUG] Add new chart heatmap causes Portal failure

### ISSUE THREAD

**Issue by MelissaGraham** [rubin-dp0/Support#46]
**Describe the bug**
When I attempt to add a new chart and select heatmap, the Portal fails to an all-grey screen, and I have to start over. This has happened over a few days with different sized query results.

**To Reproduce**
1. Use the following ADQL query via the Portal and click "Search". This part works fine.
```
SELECT coord_dec, coord_ra, 
scisql_nanojanskyToAbMag(g_calibFlux) as gmag, 
scisql_nanojanskyToAbMag(r_calibFlux) as rmag 
FROM dp02_dc2_catalogs.Object 
WHERE CONTAINS(POINT('ICRS', coord_ra, coord_dec), 
               CIRCLE('ICRS', 55.75, -32.27, 0.4)) = 1 
AND detect_isPrimary = 1 
AND g_extendedness = 1 AND r_extendedness = 1 
AND scisql_nanojanskyToAbMag(g_calibFlux) < 25 
AND scisql_nanojanskyToAbMag(r_calibFlux) < 25 
```

2. In the Results View, at upper right in the xy plot, click the double-gears icon to obtain the Plot Parameters pop-up window. Select "Add New Chart". Change the "Plot Type" to "heatmap". For "X:" enter "gmag". For "Y:" enter "gmag-rmag". Click OK.

<img width="1413" alt="Screen Shot 2023-05-11 at 6 11 41 PM" src="https://github.com/rubin-dp0/Support/assets/13826801/634ab6ad-c173-490c-883b-227b6b676f2a">

3. The screen goes all grey. The full URL of the grey screen is below. 

<img width="1268" alt="Screen Shot 2023-05-11 at 6 12 20 PM" src="https://github.com/rubin-dp0/Support/assets/13826801/9313fe59-8959-4b7c-ad8c-97b9dcf5c9bc">

```https://data.lsst.cloud/portal/app/?__action=table.search&request=%7B%22startIdx%22%3A0%2C%22serviceUrl%22%3A%22https%3A%2F%2Fdata.lsst.cloud%2Fapi%2Ftap%22%2C%22QUERY%22%3A%22SELECT%20coord_dec%2C%20coord_ra%2C%20%20scisql_nanojanskyToAbMag(g_calibFlux)%20as%20gmag%2C%20%20scisql_nanojanskyToAbMag(r_calibFlux)%20as%20rmag%20%20FROM%20dp02_dc2_catalogs.Object%20%20WHERE%20CONTAINS(POINT(%27ICRS%27%2C%20coord_ra%2C%20coord_dec)%2C%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20CIRCLE(%27ICRS%27%2C%2055.75%2C%20-32.27%2C%200.4))%20%3D%201%20%20AND%20detect_isPrimary%20%3D%201%20%20AND%20g_extendedness%20%3D%201%20AND%20r_extendedness%20%3D%201%20%20AND%20scisql_nanojanskyToAbMag(g_calibFlux)%20%3C%2025%20%20AND%20scisql_nanojanskyToAbMag(r_calibFlux)%20%3C%2025%20%22%2C%22MAXREC%22%3A50000%2C%22META_INFO%22%3A%7B%22title%22%3A%22dp02_dc2_catalogs.Object%20-%20data.lsst.cloud%2Fapi%22%2C%22tbl_id%22%3A%22tbl_id-c17-27%22%7D%2C%22tbl_id%22%3A%22tbl_id-c17-27%22%2C%22id%22%3A%22AsyncTapQuery%22%2C%22__searchID%22%3A1683852145577%2C%22pageSize%22%3A100%7D&options=%7B%22backgroundable%22%3Atrue%2C%22showFilters%22%3Atrue%2C%22showInfoButton%22%3Atrue%2C%22pageSize%22%3A100%7D```

4. Click the refresh button in the browser and the Portal returns to the results view with the default xy plot.

**Expected behavior**
Expected to be able to add a heatmap displaying g-r color vs. g-band magnitude (color-magnitude diagram).

**Desktop (please complete the following information):**
 - OS: MacOS Monterey 12.4
 - Browser: Chrome Version 112.0.5615.121

**Additional context**
I get around this by, instead of "Add New Chart", selecting "Overplot New Trace", and essentially adding the CMD to the default heatmap of RA vs. Dec. It looks like this, temporarily:

<img width="699" alt="Screen Shot 2023-05-11 at 6 17 16 PM" src="https://github.com/rubin-dp0/Support/assets/13826801/4fccb598-fcc6-4265-a20c-4e54367b2a02">

But then I remove "trace 0" (the RA vs. Dec) so I'm left with only the CMD.

**Comment by frossie** [comment]
Could you please test with the current version?

**Comment by MelissaGraham** [comment]
@frossie and @gpdf, I confirmed that steps 1 and 2 still lead to 3, the all-gray screen, unfortunately. Just now at data.lsst.cloud.

**Comment by gpdf** [comment]
Reproduced with an even more recent version, on Firefox.

**Comment by gpdf** [comment]
Reproduced in other Firefly applications.  Also confirmed that it does not depend on the plot configuration having an arithmetic expression in it (@MelissaGraham 's `gmag-rmag`).  Creating service ticket.

**Comment by gpdf** [comment]
Fixed in Firefly release-2023.1.5 and deployed to prod today.

<img width="713" alt="image" src="https://github.com/rubin-dp0/Support/assets/13261203/71ce7a27-be78-4488-ad65-eb0fed9f7568">

**Comment by gpdf** [comment]
Thank you, Trey.
